### PR TITLE
Julia Kozeeva Tasks 11-20

### PR DESCRIPTION
**Pull request by usersadvocate**

### CODE REVIEW

File: tasks/15_confirm-ending/confirm-ending.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function confirmEnding(str, target) {
+  return (str.substr(-target.length) === target) ? true : false;
```
Comment (vadeneev) [inline tasks/15_confirm-ending/confirm-ending.js]: why do we need ? : here ?

File: tasks/16_repeat-string/repeat-string.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+function repeatStringNumTimes(str, num) {
+  let newStr = '';
+  while (num > 0) {
```
Comment (vadeneev) [inline tasks/16_repeat-string/repeat-string.js]: There is much more comfortable way to achieve this goal

**Review by vadeneev — CHANGES_REQUESTED** [review]

File: tasks/19_array-chunks/array-chunks.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+function arrayChunks(arr, size) {
+  let newArray = [];
+  for (let i = 0; i < arr.length; i+=size) {
```
Comment (alexkond) [inline tasks/19_array-chunks/array-chunks.js]: There are shorter solutions without the need for iteration.

**Review by alexkond — CHANGES_REQUESTED** [review]

File: tasks/11_find-longest-word/find-longest-word.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+function findLongestWord(str) {
+  let words = str.split(' ');
```
Comment (vovcyan) [inline tasks/11_find-longest-word/find-longest-word.js]: const

File: tasks/12_title-case/title-case.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+function titleCase(str) {
+  str = str.toLowerCase();
```
Comment (vovcyan) [inline tasks/12_title-case/title-case.js]: Why do you need to change original string?

File: tasks/12_title-case/title-case.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  let words = str.split(' ');
```
Comment (vovcyan) [inline tasks/12_title-case/title-case.js]: const

File: tasks/13_largest-of-four/largest-of-four.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+function largestOfFour(arr) {
+  let largestArray = [];
```
Comment (vovcyan) [inline tasks/13_largest-of-four/largest-of-four.js]: const

File: tasks/13_largest-of-four/largest-of-four.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  for (let i = 0; i < arr.length; i++) {
+    largestArray.push(Math.max(...arr[i]));
```
Comment (vovcyan) [inline tasks/13_largest-of-four/largest-of-four.js]: Split it into different lines

File: tasks/16_repeat-string/repeat-string.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+function repeatStringNumTimes(str, num) {
+  return num > 0 ? str.repeat(num) : '';
```
Comment (vovcyan) [inline tasks/16_repeat-string/repeat-string.js:2]: Do not allow to use repeat method, please use some loop

File: tasks/17_truncate-string/truncate-string.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+function truncateString(str, num) {
+  let res = '';
```
Comment (vovcyan) [inline tasks/17_truncate-string/truncate-string.js]: const

File: tasks/17_truncate-string/truncate-string.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+  if (str.length > num) {
+    if (num > 3) {
+      res = str.slice(0, num - 3) + '...';
```
Comment (vovcyan) [inline tasks/17_truncate-string/truncate-string.js]: Please remove code duplication, combine it to one line with different params

File: tasks/19_array-chunks/array-chunks.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+function arrayChunks(arr, size) {
+  let newArray = [];
```
Comment (vovcyan) [inline tasks/19_array-chunks/array-chunks.js]: const

File: tasks/19_array-chunks/array-chunks.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+  while (arr.length) {
+    newArray.push(arr.splice(0, size));
```
Comment (vovcyan) [inline tasks/19_array-chunks/array-chunks.js]: Please split it into different lines

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/16_repeat-string/repeat-string.js (hunk 9 above)
Comment (usersadvocate) [reply tasks/16_repeat-string/repeat-string.js:2]: I had while loop, but Alex and Daniil suggest use more comfortable way  https://github.com/juliakozeeva/ui-internship-lwo2/pull/2#discussion_r323293969.
I must use if or while?

File: tasks/11_find-longest-word/find-longest-word.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+function findLongestWord(str) {
+  const words = str.split(' ');
+  let longestWord = words.reduce(function(longest, current) {
```
Comment (vovcyan) [inline tasks/11_find-longest-word/find-longest-word.js]: const

File: tasks/13_largest-of-four/largest-of-four.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+function largestOfFour(arr) {
+  const largestArray = [];
+  for (let i = 0; i < arr.length; i++) {
+    let maxItem = Math.max(...arr[i]);
```
Comment (vovcyan) [inline tasks/13_largest-of-four/largest-of-four.js]: const

File: tasks/17_truncate-string/truncate-string.js (hunk 10 above)
Comment (vovcyan) [reply tasks/17_truncate-string/truncate-string.js]: Again

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/17_truncate-string/truncate-string.js (hunk 10 above)
Comment (usersadvocate) [reply tasks/17_truncate-string/truncate-string.js]: ok, but in if statement I can't change res variable

**Review by alexkond — APPROVED** [review]

File: tasks/16_repeat-string/repeat-string.js (hunk 9 above)
Comment (vovcyan) [reply tasks/16_repeat-string/repeat-string.js:2]: Ok, got it

**Review by vovcyan — APPROVED** [review]

**Review by vadeneev — APPROVED** [review]

**Pull request merged**